File: Strings/src/main/java/Solution8.java
public class Solution8 {
    public int myAtoi(String s) {
        boolean isPre = true;
        int res = 0;
        int sign = 1;

        for (char ch : s.toCharArray()) {
            if (isPre && !checkNum(ch) && ch != '+' && ch != '-' && ch != ' ') {
                return 0;
            }

            if (!isPre && !checkNum(ch)) {
                return res * sign;
            }

            if (ch == '+' && isPre) {
                sign = 1;
                isPre = false;
            }

            if (ch == '-' && isPre) {
                sign = -1;
                isPre = false;
            }

            if (checkNum(ch)) {
                isPre = false;
                if ((res * 10L + (ch - '0')) * sign > (long) Integer.MAX_VALUE) {
                    return Integer.MAX_VALUE;
                } else if ((res * 10L + (ch - '0')) * sign < (long) Integer.MIN_VALUE) {
                    return Integer.MIN_VALUE;
                }

                res = res * 10 + (ch - '0');

            }

        }
        return res * sign;
    }

    private boolean checkNum(char ch) {
        if (ch >= '0' && ch <= '9') {
            return true;
        }
        return false;
    }

    public static void main(String[] args) {

    }
}

File: Strings/src/main/python/solution8.py
class Solution:
    def myAtoi(self, s: str) -> int:
        res = 0
        sign = 1

        is_pre = True

        for num in s:

            if (
                is_pre
                and not num.isnumeric()
                and num != "+"
                and num != "-"
                and num != " "
            ):
                return 0

            if (not is_pre) and (not num.isnumeric()):
                return res * sign

            if is_pre and num == " ":
                continue

            if num == "+" and is_pre:
                sign = 1
                is_pre = False

            if num == "-" and is_pre:
                sign = -1
                is_pre = False

            if num.isnumeric():

                is_pre = False
                res = res * 10 + int(num)

                if res * sign > 2 ** 31 - 1:
                    return 2 ** 31 - 1
                elif res * sign < -(2 ** 31):
                    return -(2 ** 31)

        return sign * res
